openmc/deplete/abc.py

```diff
@@ -701,6 +712,33 @@ def solver(self, func):
 
         self._solver = func
 
+    @property
+    def reactivity_control(self):
+        return self._reactivity_control
+
+    @reactivity_control.setter
+    def reactivity_control(self, reactivity_control):
+        if isinstance(reactivity_control, CellReactivityController):
+            if not isinstance(reactivity_control.cell, Cell):
+                raise TypeError(
+                    "Reactivity control attribute must be Cell "
+                    "not {}".format(type(reactivity_control.cell)))
+        elif isinstance(reactivity_control, MaterialReactivityController):
+            if not isinstance(reactivity_control.material, Material):
+                raise TypeError(
+                    "Reactivity control attribute must be Material "
+                    "not {}".format(type(reactivity_control.material)))
+        else:
+            raise TypeError(
+                "Reactivity Control must either be "
+                "CellReactivityController, or MaterialReactivityController, "
+                "not {}".format(type(reactivity_control)))
```

Comment:
What is the concern about only enforcing `isinstance(reactivity_control, ReactivityController)`? It's already an abstract base class so as long as someone fills out the abstract methods, any controller should work. Surely most cases will be modifying properties on cells or materials, but why forbid people from rolling their own `ReactivityController` subclass? If it's to set up protections and ensure the classes are well defined, what else can we put on the abstract base class to guide people? 

Otherwise, we would need to come back to this setter and make modifications to the source code to allow more extensions.

Recommend reading on the [Open/Closed principle of object oriented programming](https://en.wikipedia.org/wiki/Open%E2%80%93closed_principle)

> software entities (classes, modules, functions, etc.) should be open for extension, but closed for modification